modules/vebal/voting-list.service.ts

```diff
@@ -0,0 +1,188 @@
+import { AllNetworkConfigs } from '../../modules/network/network-config';
+import { networkContext } from '../../modules/network/network-context.service';
+import multicall3Abi from '../../modules/pool/lib/staking/abi/Multicall3.json';
+import {
+    LiquidityGaugesInfo,
+    RootGaugesInfo,
+    gaugeSubgraphService,
+} from '../../modules/subgraphs/gauge-subgraph/gauge-subgraph.service';
+import { getContractAt } from '../../modules/web3/contract';
+import { NetworkConfig } from '../network/network-config-types';
+import veBalHelpersAbi from './abi/veBalHelpers.json';
+import { prisma } from '../../prisma/prisma-client';
+import { Chain as PrismaChain, PrismaVotingList } from '@prisma/client';
+import { Chain as SubgraphChain } from '../subgraphs/gauge-subgraph/generated/gauge-subgraph-types';
+
+type LiquidityGaugeInfo = LiquidityGaugesInfo[number];
+interface GaugeInfo extends LiquidityGaugeInfo {
+    chain: PrismaChain;
+}
+type GaugesInfo = GaugeInfo[];
+
+export class VotingListService {
+    async syncVotingList() {
+        const promises = Object.values(AllNetworkConfigs).map(async (network) => fetchGaugesInfo(network));
+        const gaugeInfoForAllNetworks = await Promise.all(promises);
+        const gaugesInfo = flatten(gaugeInfoForAllNetworks);
+
+        console.log('Total gauges', gaugesInfo.length);
+
+        const validGauges = await filterValidGauges(gaugesInfo);
+
+        console.log('Total valid gauges', validGauges.length);
+
+        // console.log('First gauge record: ', validGauges[0]);
+        return validGauges;
+    }
+
+    async reloadVotingList(newGauges: GaugesInfo) {
+        console.log('Inserting', newGauges);
+
+        await prisma.prismaVotingList.deleteMany();
+
+        await prisma.prismaVotingList.createMany({
+            data: newGauges.map((gauge) => ({
+                address: getAddress(gauge.id),
+                chain: gauge.chain,
+                isKilled: gauge.isKilled,
+                relativeWeightCap: gauge.relativeWeightCap,
+                // poolId: gauge.poolId,
+            })),
+        });
+    }
+
+    async getList(): Promise<PrismaVotingList[]> {
+        return await prisma.prismaVotingList.findMany();
+    }
+
+    /**
+     * This query illustrates the pool related data that we need for each gauge
+     * Ideally, we don't want to denormalize but how can we join validGauges with getPoolsForVotingList in an efficient way?
+     */
+    public async getPoolsForVotingList(poolIds: string[]): Promise<any> {
+        let pools = await prisma.prismaPool.findMany({
+            where: { id: { in: poolIds } },
+            select: {
+                id: true,
+                address: true,
+                type: true, // TODO: Types are different than the original subgraph ones cause they are changed in mapSubgraphPoolTypeToPoolType (do we add a new column to prismaPool to store the original type?)
```

Comment:
In frontend-v2 we use the original types from the subgraph. This is an issue because we need different LINEAR  types [that are lost after the mapping ](https://github.com/beethovenxfi/beethovenx-backend/blob/e25ef63fd7f88b4af72fe6291f19851d32220132/modules/pool/lib/pool-creator.service.ts#L413).
In the long term, we should probably try to find a common shared set of types for both frontends. 
In the short term, should we add a column "originalPoolType" or "balancerPoolType" in the tables using it?

Reply:
Why do you need the different linear types? Why do you need the pooltypes at all?

Reply:
We don't need them for this concrete feature, so let's forget this for now (probably YAGNI)